c7n/resources/appelb.py

```diff
@@ -307,24 +307,49 @@ def initialize(self, albs):
 
 @filters.register('listener')
 class AppELBListenerFilter(ValueFilter, AppELBListenerFilterBase):
-    """Filter ALB based on matching listener attributes"""
+    """Filter ALB based on matching listener attributes
 
-    schema = type_schema('listener', rinherit=ValueFilter.schema)
+    Adding the `matched` flag will filter on previously matched listeners
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: app-elb-invalid-ciphers
+                resource: app-elb
+                filters:
+                  - type: listener
+                    key: Protocol
+                    value: HTTPS
+                  - type: listener
+                    key: SslPolicy
+                    value: ['ELBSecurityPolicy-TLS-1-1-2017-01','ELBSecurityPolicy-TLS-1-2-2017-01']
+                    op: ni
+                    matched: true
+                actions:
+                  - type: modify-listener
+                    sslpolicy: "ELBSecurityPolicy-TLS-1-2-2017-01"
+    """
+
+    schema = type_schema('listener', rinherit=ValueFilter.schema, matched={'type': 'boolean'})
     permissions = ("elasticloadbalancing:DescribeLoadBalancerAttributes",)
 
     def process(self, albs, event=None):
         self.initialize(albs)
         return super(AppELBListenerFilter, self).process(albs, event)
 
     def __call__(self, alb):
-        matched = []
-        for listener in self.listener_map[alb['LoadBalancerArn']]:
+        listeners = self.listener_map[alb['LoadBalancerArn']]
+        if self.data.get('matched', False):
```

Comment:
do we even need to initialize if matched is true? also use of matched should have corresponding validate method  check to verify that usage is subsequent to an extant listener filter.